pkg/imageregistry/imageregistry.go

```diff
@@ -0,0 +1,227 @@
+package imageregistry
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	imageregistryv1 "github.com/openshift/api/imageregistry/v1"
+	v1 "github.com/openshift/api/operator/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// Builder provides a struct for imageRegistry object from the cluster and a imageRegistry definition.
+type Builder struct {
+	// imageRegistry definition, used to create the imageRegistry object.
+	Definition *imageregistryv1.Config
+	// Created imageRegistry object.
+	Object *imageregistryv1.Config
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// Used in functions that define or mutate clusterOperator definition. errorMsg is processed before the
+	// ClusterOperator object is created.
+	errorMsg string
+}
+
+// Pull retrieves an existing imageRegistry object from the cluster.
+func Pull(apiClient *clients.Settings, imageRegistryObjName string) (*Builder, error) {
+	if apiClient == nil {
+		glog.V(100).Infof("The apiClient is empty")
+
+		return nil, fmt.Errorf("imageregistry Config 'apiClient' cannot be empty")
+	}
+
+	glog.V(100).Infof(
+		"Pulling imageRegistry object name: %s", imageRegistryObjName)
+
+	builder := Builder{
+		apiClient: apiClient,
+		Definition: &imageregistryv1.Config{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: imageRegistryObjName,
+			},
+		},
+	}
+
```

Comment:
Oh sorry missed this one.
Can you please add argument validation?

```
	if imageRegistryObjName == "" {
		glog.V(100).Infof("The name of the imageRegistry is empty")

		return nil, fmt.Errorf("imageRegistry 'imageRegistryObjName' cannot be empty")
	}
```

Reply:
sure, thank you! fixed